### PR TITLE
Location trigger - added option for Going inside/outside of area 

### PR DESCRIPTION
**Pull request by PoweRGbg**

### CODE REVIEW

File: app/src/main/java/info/nightscout/androidaps/plugins/general/automation/triggers/TriggerLocation.java [hunk 1]
```diff
@@ -71,8 +76,8 @@ public synchronized boolean shouldRun() {
         a.setLatitude(latitude.getValue());
         a.setLongitude(longitude.getValue());
         double calculatedDistance = location.distanceTo(a);
-
-        if (calculatedDistance < distance.getValue()) {
+        if (((modeSelected.getValue().ordinal()) == 1d) && (calculatedDistance < distance.getValue()) ||
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/automation/triggers/TriggerLocation.java]: use value instead of ordinal

Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/automation/triggers/TriggerLocation.java]: <=

File: app/src/main/java/info/nightscout/androidaps/plugins/general/automation/elements/InputLocationMode.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        GOING_IN,
+        GOING_OUT;
+
+        public @StringRes
+        int getStringRes() {
+            switch (this) {
+                case INSIDE:
+                    return R.string.location_inside;
+                case OUTSIDE:
+                    return R.string.location_outside;
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/automation/elements/InputLocationMode.java]: on 2 are implemented now?

**Comment by PoweRGbg** [comment]
Closing for more testing

**Comment by PoweRGbg** [comment]
Ready for public testing? Still too many debugging and when AandroidAPS initializes LocationService returns 0 values which causes automation to fire-up GOING_OUT & GOING_IN

**Comment by MilosKozak** [comment]
`InputOption` can be removed or?

**Comment by PoweRGbg** [comment]
> `InputOption` can be removed or?

Yes actually. It is not used in this trigger.

**Pull request merged**